addons/Paragraph/src/presenter.js

```diff
@@ -833,29 +920,26 @@ function AddonParagraph_create() {
         $paragraph.css("border", "1px solid");
         $paragraph.css("padding", "10px");
 
-        if (showAnswers && modelAnswer && !usersAnswer) {
+        if (showAnswers && modelAnswer && !presenter.didUserAnswer(usersAnswer)) {

```

Comment:
Ten warunek powoduje, że gdy wywołujemy GetPrintableHTML i chcemy uzyskać drukowalną wersję z poprawnymi odpowiedziami to jeśli w jakimś addonie użytkownik udzielił odpowiedzi to dostaniemy pomieszaną wersję z poprawnymi odpowiedziami tam gdzie odpowiedź była pusta, a paragrafy gdzie użytkownik udzielił odpowiedzi będą zawierały odpowiedź użytkownika zamiast poprawnej wersji. 

W przypadku tego zadania getPrintableHTML powinno pracować jedynie w 2 trybach: 
A) Wszystkie paragrafy są puste
B) Wszystkie paragrafy są uzupełnione poprawnymi odpowiedziami

W drukowalnej wersji nie mieszamy tych 2 wersji. Dopiero gdy mamy zadanie z printowalną wersją to odpowiedzi użytkownika są ustawiane przy pomocy setState.

Reply:
Zgodnie z daily są dwa rozwiązania:
A) Dodanie setPrintableState i na podstawie tego zaimplementować obsługę 4 stanów:
- Empty paragraph
- Show answer
- Show user answer
- Check user answers